checkstyle-tester/diff.groovy

```diff
@@ -365,14 +365,72 @@ def cloneRepository(repoName, repoType, repoUrl, commitId, srcDir) {
         def lastCommitSha = getLastProjectCommitSha(repoType, srcDestinationDir)
         def commitIdSha = getCommitSha(commitId, repoType, srcDestinationDir)
         if (lastCommitSha != commitIdSha) {
-            def resetCmd = getResetCmd(repoType, commitId)
-            println "Resetting $repoType sources to commit '$commitId'"
-            executeCmd(resetCmd, new File("$srcDestinationDir"))
+            if (isGitSha(commitId)) {
+                // If commitId is a SHA, reset directly
+                def resetCmd = getResetCmd(repoType, commitId)
+                println "Resetting $repoType sources to commit '$commitId'"
+                executeCmd(resetCmd, new File("$srcDestinationDir"))
+            } else {
+                // If commitId is a branch or tag, fetch more data and then reset
+                fetchAdditionalData(repoType, srcDestinationDir, commitId)
+                def resetCmd = getResetCmd(repoType, commitId)
+                println "Resetting $repoType sources to commit '$commitId'"
+                executeCmd(resetCmd, new File("$srcDestinationDir"))
+            }
```

Comment:
Can we avoid repetition here?
```suggestion
            if (!isGitSha(commitId)) {
                // If commitId is a branch or tag, fetch more data and then reset
                fetchAdditionalData(repoType, srcDestinationDir, commitId)
            } 
                def resetCmd = getResetCmd(repoType, commitId)
                println "Resetting $repoType sources to commit '$commitId'"
                executeCmd(resetCmd, new File("$srcDestinationDir"))
```

Reply:
Yes, thank you for highlighting. Refactored.

Reply:
Yes, thank you for highlighting. Refactored.